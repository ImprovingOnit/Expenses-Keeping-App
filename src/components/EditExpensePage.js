import React from 'react'



const EditExpensePage = () => {
    return (
        <div>
            Edit Page
        </div>
    )
}

export default EditExpensePage