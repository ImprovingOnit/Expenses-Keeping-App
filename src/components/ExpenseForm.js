import React from 'react'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'

class ExpenseForm extends React.Component {

    state = { 
        description: this.props.expense ? this.props.expense.description : '',
        note: this.props.expense ? this.props.expense.note : '',
        amount: this.props.expense ? (this.props.expense.amount / 100).toString() : '',
        createdAt: this.props.expense ? moment(this.props.expense.createdAt) : moment(),
        calendarFocused: false,
        error: ''
    }

    onInputChangeDescription = (e) => {
        const description = e.target.value
        this.setState({ description })
    }

    onTextaraeChangeNote = (e) => {
        const note = e.target.value
        this.setState({ note})
    }

    onInputChangeAmount = (e) => {
        const amount = e.target.value
        if (!amount || amount.match(/^\d*(\.\d{0,2})?$/)) {
            this.setState({ amount })
        }
    }

    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState({ createdAt })
        }
    }

    onFocusChange = ({ focused }) => {
        this.setState({ calendarFocused: focused })
    }

    onFormSubmit = (e) => {
        e.preventDefault()

        if (!this.state.description || !this.state.amount) {
            this.setState({ error: 'Please provide description and amount' })
        } else {
            this.setState({ error: '' })
            this.props.onFormSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            })
        }
    }

    render () {
        return (
            <form className="form" onSubmit={this.onFormSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>} 
                <input 
                    type="text" 
                        placeholder="Description"
                            value={this.state.description}
                                autoFocus
                                    onChange={this.onInputChangeDescription}
                                        className="text-input"
                />
                <input 
                    type="text"
                        placeholder="Amount"
                            value={this.state.amount}
                                onChange={this.onInputChangeAmount}
                                    className="text-input"
                />
                <SingleDatePicker 
                    date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                            focused={this.state.calendarFocused}
                                onFocusChange={this.onFocusChange}
                                    numberOfMonths={1}
                                        isOutsideRange={() => false}
                />
                <textarea
                    placeholder="Add a note for your expense"
                        onChange={this.onTextaraeChangeNote}
                            className="textarea"
                >
                </textarea>
                <div>
                    <button className="button">{this.props.expense ? 'Edit Expense' : 'Add Expense'}</button>
                </div>
            </form>
        )
    }
}


export default ExpenseForm