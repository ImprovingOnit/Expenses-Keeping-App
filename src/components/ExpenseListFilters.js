import React from 'react'
import { connect } from 'react-redux';
import { setTextFilter } from './../actions/filters';

const ExpenseListFilters = (props) => {

    return (
        <div>
            <input 
                type="text" 
                    onChange={(e) => props.setTextFilterOnRedux(e.target.value)} 
                        value={props.filters.text}/>
        </div>
    )
}

const mapStateToProps = state => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = dispatch => {
    return {
        setTextFilterOnRedux: (text) => dispatch(setTextFilter(text))
    }
}

export default  connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)