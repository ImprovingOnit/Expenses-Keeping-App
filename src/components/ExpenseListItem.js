import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'



export const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (
        <Link 
            to={`/edit/${id}`}
                className="list-item"
        >
            <div>
                <h2 className="list-item__title">{description}</h2>
                <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YY')}</span>
            </div>
            <h2 className="list-item__data">
                {numeral(amount / 100).format('$0,0.00')}
            </h2>
        </Link>
    )
}



export default ExpenseListItem