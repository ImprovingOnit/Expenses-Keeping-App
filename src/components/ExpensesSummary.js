import React  from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import expensesTotal from '../selectors/expenses-total'
import selectExpense from '../selectors/expenses'

export const ExpensesSummary = (props) => {
    return (
        <div>
            {
            props.expenseCount <= 1 ? (
                <h1>Viewing {props.expenseCount} expense totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</h1>
                ) : (
                <h1>Viewing {props.expenseCount} expenses totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</h1>
                )
            }
        </div>
    )
}

const mapStateToProps = state => {
    const visibleExpenses = selectExpense(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)