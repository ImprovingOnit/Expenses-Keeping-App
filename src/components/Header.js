import React from 'react'
import { NavLink } from 'react-router-dom'


const Header = () => {
    return (
        <div>
            <h1>Expensify</h1>
            <NavLink exact activeClassName="is-active" to="/">Home Page</NavLink>
            <NavLink activeClassName="is-active" to="/create">AddExpense</NavLink>
            <NavLink activeClassName="is-active" to="/help">Help</NavLink>
        </div>
    )
}

export default Header