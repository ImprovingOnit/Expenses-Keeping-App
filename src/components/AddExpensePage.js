import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { addExpense } from './../actions/expenses'

export const AddExpensePage = (props) => {
    return (
        <div>
            <h1>AddExpense</h1>
            <ExpenseForm 
                onFormSubmit={(expense) => {
                    props.addExpenseToRedux(expense)
                    props.history.push('/')
                }}
            />
        </div>
    )
}

const mapDispatchToProps = dispatch => {
    return {
        addExpenseToRedux: expense => dispatch(addExpense(expense))
    }
}

export default connect(null, mapDispatchToProps)(AddExpensePage)