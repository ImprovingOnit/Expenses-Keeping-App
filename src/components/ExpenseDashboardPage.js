import React from 'react'

import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardPage = () => {
    return (
        <div>
            DashBoard
            <ExpenseListFilters />
            <ExpenseList />
        </div>
    )
}

export default ExpenseDashboardPage
