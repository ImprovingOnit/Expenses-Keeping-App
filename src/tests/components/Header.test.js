import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Header } from './../../components/Header'

test('render Header component', () => {
    const wrapper = shallow(<Header startLogout={() => {}}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('call startLogout on button click', () => {
    const startLogout = jest.fn()
    const wrapper = shallow(<Header startLogout={startLogout} />)
    wrapper.find('button').simulate('click')
    expect(startLogout).toHaveBeenCalled()
})