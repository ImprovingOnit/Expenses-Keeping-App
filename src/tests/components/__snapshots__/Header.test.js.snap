// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header component 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Home Page
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    AddExpense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
